### PR TITLE
Search for anaconda installs of retriever

### LINKED ISSUE #70: won't work with RStudio + anaconda python
I'm running into the same issue as #10 and #53  

I'm running Rstudio and miniconda on Ubuntu 15.10. I've traced it back to the `Sys.which('retriever')` call in `ecoretriever.R`. 

When I run it in Rstudio it fails because it can't find retriever in my local `~/miniconda3/bin` path

```
> Sys.which('retriever')
retriever 
       "" 
> system('echo $PATH')
/home/shawn/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/shawn/bin//bin
> 
```

Even though it works with R from the command line

```
> Sys.which('retriever')
                             retriever 
"/home/shawn/miniconda3/bin/retriever" 
> system('echo $PATH')
/home/shawn/miniconda3/bin:/home/shawn/anaconda3/bin:/home/shawn/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/shawn/bin//bin
> 
```

Looks like RStudio does not get access to user configured paths. So it can't find `retriever` installed in my `~/miniconda` directory (same would be true for anaconda). This is an issue with [other](https://support.rstudio.com/hc/en-us/community/posts/203561546-RStudio-can-t-detect-pdflatex-No-Tex-installation-detected-) software too. I'll look into this more when I have time.

### PR DESCRIPTION
**Pull request by sdtaylor**
Fixes #70 by adding the default anaconda/miniconda install directories to the search path. Tested on Ubuntu. Still needs to be tested on Windows and Mac before merging.

### CODE REVIEW

**Comment by dmcglinn** [comment]
Thanks for adding these changes @sdtaylor! I only have a few small suggestions. When I tested your code on Windows I got an error b/c of the forward slashes in the pathes. I propose therefore this version of the function (see below) which is a bit simpler and easier to read. Also it its ok with you I'll add you to the list of package coauthors. Let me know what you think. 
```r
check_for_retriever = function(...) {
    retriever_path = Sys.which('retriever')
    
    #Rstudio will not import any paths configured for anaconda python installs, so add default anaconda paths
    #manually. See http://stackoverflow.com/questions/31121645/rstudio-shows-a-different-path-variable
    if (retriever_path == '') {
        os = Sys.info()[['sysname']]
        home_dir = dirname(Sys.getenv('HOME'))
        possible_pathes = c('/Anaconda3/Scripts',
                            '/Anaconda2/Scripts',
                            '/Miniconda3/Scripts',
                            '/Miniconda2/Scripts',
                            '/anaconda3/bin',
                            '/anaconda2/bin',
                            '/miniconda3/bin',
                            '/miniconda2/bin')
        for (i in possible_pathes) {
            Sys.setenv(PATH = paste(Sys.getenv('PATH'), ':', home_dir, i), sep='')
        }      
    }
    
    retriever_path = Sys.which('retriever')
    
    if (retriever_path == '') {
        path_warn = 'The retriever is not on your path and may not be installed.'
        mac_instr = 'Follow the instructions for installing and manually adding the EcoData Retriever to your path at http://ecodataretriever.org/download.html'
        download_instr = 'Please upgrade to the most recent version of the EcoData Retriever, which will automatically add itself to the path http://ecodataretriever.org/download.html'
        os = Sys.info()[['sysname']]
        if (os == 'Darwin')
            packageStartupMessage(paste(path_warn, mac_instr))
        else 
            packageStartupMessage(paste(path_warn, download_instr))
    }    
}
```

**Comment by dmcglinn** [comment]
Oh and I got rid of the function `paste0` in favor of traditional `paste` because `paste0` is only available in newer versions of R and it doesn't seem worth it to lose compatibility with older versions of R.

**Comment by sdtaylor** [comment]
Thanks Dan, these changes look good to me. I'll try to get in front of a windows computer to test them out soon.

**Comment by dmcglinn** [comment]
@ethanwhite I couldn't figure out to how push my commits on @sdtaylor's PR back to this thread. I assume that is what you mean by "update the PR". If you wouldn't mind could you share the workflow you use to do this?

**Comment by ethanwhite** [comment]
@dmcglinn - if you want to make improvements to someone else's PR prior to any merging going on then one option is to make a PR into their repo on the branch they made the PR from. If they merge that PR then those changes will cascade through. There is also a newish default behavior in GitHub that allows maintainers to directly add commits to the contributing branch. I only know how to do this in the browser at the moment. If you click on the `Files changed` tab and then click on the edit button (pen(cil) icon) for the file you want to change you can make edits in the browser and the commit them directly.

**Comment by dmcglinn** [comment]
Sweet thanks for the info!

**Comment by ethanwhite** [comment]
I just updated this PR with @dmcglinn's suggested changes. This appears to be working on both Windows 10 and Ubuntu for me.

**Comment by ethanwhite** [comment]
OK, so this fixes Windows but breaks *nix unfortunately.

File: R/ecoretriever.R [hunk 1]
```diff
@@ -206,19 +206,20 @@ check_for_retriever = function(...) {
...
+        possible_pathes = c('/Anaconda3/Scripts',
+                            '/Anaconda2/Scripts',
+                            '/Miniconda3/Scripts',
+                            '/Miniconda2/Scripts',
+                            '/anaconda3/bin',
+                            '/anaconda2/bin',
+                            '/miniconda3/bin',
+                            '/miniconda2/bin')
+        for (i in possible_pathes) {
+            Sys.setenv(PATH = paste(Sys.getenv('PATH'), ':', home_dir, i), sep='')
```
Comment (sdtaylor) [inline R/ecoretriever.R]: There is a misplaced `)` here giving issues with ubuntu.
```
Sys.setenv(PATH = paste(Sys.getenv('PATH'), ':', home_dir, i, sep='')) 
```

**Comment by ethanwhite** [comment]
OK, #80 appears to be working on both Windows 10 and Ubuntu. Give it a test if you get the chance.

**Pull request closed without merging**